include/exadg/aero_acoustic/driver.cpp

```diff
@@ -188,18 +197,28 @@ Driver<dim, Number>::print_performance_results(double const total_time) const
   pcout << std::endl << "Timings for level 2:" << std::endl;
   timer_tree.print_level(pcout, 2);
 
+  // Throughput in DoFs/s per time step per core (during the time both
+  // solvers ran side by side)
+  dealii::Utilities::MPI::MinMaxAvg time_solvers_side_by_side_data =
+    dealii::Utilities::MPI::min_max_avg(time_solvers_side_by_side, mpi_comm);
+  double const time_solvers_side_by_side_avg = time_solvers_side_by_side_data.avg;
+
+  dealii::types::global_dof_index const DoFs =
+    fluid->pde_operator->get_number_of_dofs() +
+    acoustic->get_average_sub_time_steps() * acoustic->pde_operator->get_number_of_dofs();
```

Comment:
Related to comments https://github.com/exadg/exadg/pull/645/files#r1561011435 and https://github.com/exadg/exadg/pull/645/files#r1561016787, I asked myself if we should actually calculate the throughput like this. If the number of acoustic sub steps is very large, the throughput is dominated by the acoustic problem, right? But I am not sure if this is the intuitive interpretation of "throughput" from a user perspective. From such a user perspective, I am not sure if I would actually like to see the throughput per macro time step with the acoustic DoFs counted only once. So I am arguing that choosing sub-stepping is a detail of the implementation (as an alternative to implicit time stepping). A high throughput should in my opinion indicate a fast computation per macro time step, but is this really the case if e.g. the fluid solver is slow but the number of acoustic sub steps is very large?

Reply:
This is still open. Maybe let me reformulate this comment: 

Current throughput definition: $T = \frac{N_{DoF, fluid} + n_{sub} N_{Dof, acoustic}}{t_{fluid} + t_{acoustic}} = \frac{N_{DoF, fluid}/n_{sub} + N_{Dof, acoustic}}{\frac{t_{fluid} + t_{acoustic}}{n_{sub}}}$

Doesn't this mean we currently compute the throughput of a single acoustic sub step? From an application perspective, I would currently argue we should compute the throughput of a macro time step $T = \frac{N_{DoF, fluid} + N_{Dof, acoustic}}{t_{fluid} + t_{acoustic}}$.

One can maybe argue for both variants. However, I am not sure if the screen output "Throughput while both solvers ran side by side:" is correct right now. For the present implementation, I would call this quantity "Throughput related to one acoustic sub step.".

There would be yet another solution, namely to simply compute and print both metrics.

What do you think?

Reply:
Thank you for the clarification. I agree in parts. I would say, the first definition should be related to the macro time step and the second should be related to the sub time step. I.e., $N_{DoF,f}+n_{sub}N_{DoF,a}$ is the number of DoFs that is solved (in average) during one macro time step while $N_{DoF,f}+N_{DoF,a}$ DoFs are solved during one sub time step. 

I added the corresponding lines in [222c01b](https://github.com/exadg/exadg/pull/645/commits/222c01bac0b8ad70844d6f5b71110f27ffc215ca). In case you do not agree, could you explain where you think above reasoning is not correct? :)

Since $n_{sub}$ is in our case the average number of sub time steps per macro time step, the average number of DoFs per macro time step are represented by a floating point number. To account for this, I added an overload  for the function that prints the stats [05b188d](https://github.com/exadg/exadg/pull/645/commits/05b188db6abe7a93d73e9c6b4a9d376c335b3247).

Reply:
I think it should be the other way round. The first equation in my comment above shows the throughput related to a sub-step (where I have added the second version of this equation in order to highlight that this refers indeed to a sub-step). I don't agree that $N_{Dofs, f} + N_{Dofs, a}$ are the DoFs solved during one sub time step. This would be $N_{Dofs, f}/n_{sub} + N_{Dofs, a}$, and in this case one has to divide the wall time by $n_{sub}$. So the two versions of the first equation are equivalent and just reflect "different perspectives".

The second equation in my comment above refers to a macro time step, where the sub-stepping done for acoustics is kind of an irrelevant detail. Hence, acoustic dofs are only counted once. Think of using implicit time stepping for acoustics as an alternative. A reasonable throughput definition should allow in my opinion to compare these two timestepping variants quantitatively (higher throughput = faster solution for the same macro time step size), meaning they have to use the same throughput definition (related to one macro time step). The first equation instead refers to a sub-step and, hence, does not allow a reasonable comparison of these two timestepping variants. It is a metric that is only relevant for the "special case" using sub-stepping. So maybe it helps if you think how to implement the printing of the throughput metrics for these two time stepping variants and which if-statements we would need for both cases. Put differently, we can look which of the equations depends on the parameter $n_{sub}$ and which does not.

Reply:
Thank you for the clarification. I think I finally understood your reasoning :) [6a024cd](https://github.com/exadg/exadg/pull/645/commits/6a024cd4c59a5c7d5e2266b34be6adc6e621e569)